torch/_subclasses/meta_utils.py

```diff
@@ -791,40 +787,61 @@ def empty_create_subclass(
             #
             # Morally, the code here is same as transform_subclass, but we've
             # written it from scratch to read EmptyCreateSubclass
-
             outer_size = outer_size if outer_size is not None else t.size
             outer_stride = outer_stride if outer_stride is not None else t.stride
 
-            def transform(attr, inner_t):
-                r = callback(
-                    lambda: empty_create(
-                        inner_t,
-                        AttrSource(source, attr),
-                        symbolic_context=(
-                            None
-                            if symbolic_context is None
-                            else symbolic_context.inner_contexts[attr]
-                        ),
-                    )
-                )
-                if self.copy_data:
-                    with torch.no_grad(), no_dispatch():
-                        r.real_tensor = torch.empty_strided(
-                            inner_t.size,
-                            inner_t.stride,
-                            dtype=inner_t.dtype,
-                            device=inner_t.device,
+            assert symbolic_context is None or isinstance(
+                symbolic_context, SubclassSymbolicContext
+            )
+
+            def _empty_create_subclass(
+                t, outer_size, outer_stride, symbolic_context, callback, source
+            ):
+                # We are hitting plain meta_desc tensor so actually
+                # create a tensor here.
+                if t.attrs is None:
+                    r = callback(
+                        lambda: empty_create(
+                            t,
+                            source,
+                            symbolic_context,
                         )
-                        assert inner_t.data is not None
-                        _safe_copy(r.real_tensor, inner_t.data)
-                return r
+                    )
+                    if self.copy_data:
+                        with torch.no_grad(), no_dispatch():
+                            r.real_tensor = torch.empty_strided(
+                                t.size,
+                                t.stride,
+                                dtype=t.dtype,
+                                device=t.device,
+                            )
+                            assert t.data is not None
+                            _safe_copy(r.real_tensor, t.data)
+                    return r
+
+                inner_tensors = {}
+                for attr, meta_tensor_desc in t.attrs.items():
+                    current_context = None
+                    if symbolic_context is not None:
+                        current_context = symbolic_context.inner_contexts[attr]
+
+                    current_source = AttrSource(source, attr)
+                    new_empty_tensor = _empty_create_subclass(
```

Comment:
You don't have to fix it here, but there's a somewhat prevalent antipattern in this file of doing small recursions on helper functions, rather than calling all the way back to the very top level. I think it should be OK to recurse to the very top call function, and that makes things more general since you can handle composition of things with other things the small helpers don't help. Just calling attention to this.